cyclonedx/model/release_note.py

```diff
@@ -0,0 +1,222 @@
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) OWASP Foundation. All Rights Reserved.
+import datetime
+from typing import List, Optional
+
+from ..model import IssueType, Note, Properties, XsUri
+
+
+class ReleaseNotes:
+    """
+    This is our internal representation of a `releaseNotesType` for a Component in a BOM.
+
+    .. note::
+        See the CycloneDX Schema definition: https://cyclonedx.org/docs/1.4/#type_releaseNotesType
+    """
+
+    def __init__(self, type: str, title: Optional[str] = None, featured_image: Optional[XsUri] = None,
+                 social_image: Optional[XsUri] = None, description: Optional[str] = None,
+                 timestamp: Optional[datetime.datetime] = None, aliases: Optional[List[str]] = None,
+                 tags: Optional[List[str]] = None, resolves: Optional[List[IssueType]] = None,
+                 notes: Optional[List[Note]] = None, properties: Optional[Properties] = None) -> None:
+        self._type: str = type
+        self._title: Optional[str] = title
+        self._featured_image: Optional[XsUri] = featured_image
+        self._social_image: Optional[XsUri] = social_image
+        self._description: Optional[str] = description
+        self._timestamp: Optional[datetime.datetime] = timestamp
+        self._aliases: Optional[List[str]] = aliases or None
+        self._tags: Optional[List[str]] = tags or None
+        self._resolves: Optional[List[IssueType]] = resolves or None
+        self._notes: Optional[List[Note]] = notes or None
+        self._properties: Optional[Properties] = properties or None
+
+    def add_alias(self, alias: str) -> None:
+        """
+        Adds an alias to this Release.
+
+        An alias is an alternate name the release may be referred to. This may include unofficial terms used by
+        development and marketing teams (e.g. code names).
+
+        Args:
+            alias:
+                `str` alias
+        """
+        self._aliases.append(alias)
+
+    def add_tag(self, tag: str) -> None:
+        """
+        Adds a tag to this Release.
+
+        Tags may aid in search or retrieval of the release note.
+
+        Args:
+            tag:
+                `str` tag to add
+        """
+        self._tags.append(tag)
+
+    def add_resolves(self, issue: IssueType) -> None:
+        """
+        Adds an issue that this Release resolves.
+
+        Args:
+            issue:
+                `IssueType` object that is resolved by this Release
+        """
+        self._resolves.append(issue)
+
+    def add_note(self, note: Note) -> None:
+        """
+        Adds a Release Note to this Release.
+
+        Multiple note elements may be specified to support release notes in a wide variety of languages.
+
+        Args:
+            note:
+                `Note` to be added
+        """
+        self._notes.append(note)
+
+    def get_type(self) -> str:
```

Comment:
why not use python's `@propery` for all those getters? 
see https://pythonguide.readthedocs.io/en/latest/python/property.html#property